components/signup/Auth/index.js

```diff
@@ -0,0 +1,129 @@
+import React, { useCallback, useState, useEffect, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import Link from 'next/link';
+
+import GoogleLogo from 'assets/svg/btn_google_light_normal_ios.svg';
+import { emailRule } from '~/lib/regex';
+import useInput from '~/hooks/useInput';
+import { getVerifyEmailLoading } from '~/redux/selectors/signup';
+import { landingEmailSelector } from '~/redux/selectors/landing';
+import { verifyEmailRequest } from '~/redux/reducers/signup';
+import { saveEmail as saveLandingEmail } from '~/redux/reducers/landing';
+
+import * as CS from '../common/styles';
+import * as S from './style';
+
+const Index = () => {
+  const dispatch = useDispatch();
+  const submitEmailButtonRef = useRef(null);
+  const [email, onChangeEmail, setEmail] = useInput('');
```

Comment:
```suggestion
  const emailInputRef = useRef('');
```
email 상태에 대해 onChange 발생 시 리렌더링을 일으켜야 할 이유가 있나요?
아니라면 ref 로 관리하는게 성능 상 이점이 매우 클거에요.

Reply:
useInput 에서 리턴하는 onChangeEmail 은 setEmail 에 단순히 useCallback 을 감싼 형태인가요?
그렇다면 연산 비용이 이렇게 될거에요.
onChangeEmail > setEmail